SchoolProject/src/main/java/Main.java

```diff
@@ -1,51 +1,113 @@
-        System.out.println("Welcome to " + school.getName());
-        System.out.println("=============================");
+        //Add data to repositories
+        WriteDataService writeDataService = new WriteDataService();
+        writeDataService.writeData(students,studentRepository);
+        writeDataService.writeData(teachers,teacherRepository);
+        for (Teacher teacher : teachers) {
+            teacher.getSubjectList().forEach(subjectRepository::save);
+        }
+        //See data in repositories
+        //view.showMessage(studentRepository.findAll().toString());
+        //view.showMessage(teacherRepository.findAll().toString());
+        //view.showMessage(subjectRepository.findAll().toString());
+
+        //Create school
+        var school = new School("American School");
 
-        System.out.println("Select your role please: ");
-        System.out.println("1. Teacher");
-        System.out.println("2. Student");
+        view.showMenu(school.getName());
+        String option = view.getString();
 
-        String option = scanner.nextLine();
         boolean access = false;
 
         while (!access) {
-            System.out.println("Please enter your code");
-            AccessValidator accessValidator = new AccessValidator(studentRepository);
+            view.showMessage("Please enter your code");
+
 
-            String code = scanner.nextLine();
+            String code = view.getString();
 
             switch (option) {
                 case "1":
+                    var accessValidator = new AccessValidator(teacherRepository);
                     var teacher = accessValidator.verifyTeacherAccess(code);
                     if (teacher != null) {
                         access = true;
+                        view.showMessage("Welcome " + teacher.getName());
+                        //List the students of "ELEMENTARY_SCHOOL" grade
+                        //Enter a grade for a student
+                        view.showMessage("1. List the students of a grade");
+                        view.showMessage("2. Enter a grade for a student");
+
+                        String option2 = view.getString();
+
+                        switch (option2) {
+                            case "1":
+                                view.showMessage("Enter the grade");
+                                String grade = view.getString();
+                                view.showMessage(studentRepository.findByGrade(grade).toString());
+                                break;
+                            case "2":
+                                view.showMessage("Enter the student code");
+                                String studentCode = view.getString();
+                                view.showMessage("Enter the grade");
+                                String grade2 = view.getString();
+                                var x =  studentRepository.findByCode(studentCode).get();
```

Comment:
Maybe you can create a StudentService class as we have for the unit tests projects, this is because in the repository we have the main information about students, and in order to protect the information it is good to have an intermediary class to have access to that info. (it is not a rest application but we can adapt it)